app/services/foreman/deprecation.rb

```diff
@@ -0,0 +1,21 @@
+module Foreman
+  class Deprecation
+    #aviliable options: controller, method, general info, callstack
+    def self.deprecation_warning(foreman_version_deadline, options={})
+      raise Foreman::Exception, _("Invalid version format, please enter in x.y.z") unless foreman_version_deadline.to_s.match(/\A^\d[.]\d+[.]\d$\z/)
+      controller = options[:controller]
+      method = options[:method]
+      info = options[:info]
+      callstack = options[:callstack] || caller
+      msg = _("You are using deprecated behavior, it will be removed in version %{version_deadline}.\n") % {:version_deadline => foreman_version_deadline}
+      msg +=_("Use %{controller}#%{method} instead.\n") % {:controller => controller, :method => method} if controller && method
```

Comment:
There's a period after each sentence, they aren't really connected to one another so i don't know if it will help the translation?